host/fs/ioctl_mips_like.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// +build mips mipsle
+
+package fs
+
+func translateOp(op uint) uint {
+	// Decode the arm/x64 encoding and reencode as MIPS specific linux ioctl.
+	// arm/x64: DIR(2), SIZE(14), TYPE(8), NR(8)
+	// mips:    DIR(3), SIZE(13), TYPE(8), NR(8)
+	const mask = 1<<(14+8+8) - 1
+	dir := op &^ mask
+	if (dir & 1 << (13 + 8 + 8)) != 0 {
```

Comment:
Haven't you just cleared all but the dir bits in the line above? I guess, this `if` would always be false.

Reply:
You are right, the code was particularly broken. Sorry about that. I moved the code out so I could test it unconditionally.

Reply:
Please take a look at the new commit.